user/permissions.py

```diff
@@ -0,0 +1,12 @@
+from rest_framework.permissions import SAFE_METHODS, BasePermission
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        return bool(
+            request.method in SAFE_METHODS and
+            request.user and
+            request.user.is_authenticated
+        ) or (
+                request.user and request.user.is_staff
+        )
```

Comment:
The logic in `has_permission` could be simplified for clarity. Consider checking if the user is authenticated first, then checking if they are a staff member. This would make the intent of the permission class clearer.